### PR TITLE
drop python2, wrap repetitive code, minor edits

### PR DESCRIPTION
**Pull request by vkhodygo**
Hi @cjekel 

I was reading the code because I couldn't find some features in the documentation (they're there but too deep) and I noticed that some identical pieces are used through the code.
I tried to make some simple functions to improve overall readability.
I also did some minor editing, part of it is not compatible with python 2.7, don't you think it's time to drop it completely?

P.S. funny thing, while I was looking for your repo I found  [these](https://github.com/google/pwlfit) shameless folks

### CODE REVIEW

File: pwlf/pwlf.py [hunk 1]
```diff
@@ -189,39 +188,29 @@ def __init__(self, x, y, disp_res=False, lapack_driver='gelsd', degree=1,
...
+        self.break_0, self.break_n = np.min(self.x_data), np.max(self.x_data)
 
-        if degree < 12 and degree >= 0:
+        if 12 > degree >= 0:
             # I actually don't know what the upper degree limit should
             self.degree = int(degree)
         else:
-            not_suported = "degree = " + str(degree) + " is not supported."
-            raise ValueError(not_suported)
+            raise ValueError(f"degree = {degree} is not supported.")
```
Comment (cjekel) [inline pwlf/pwlf.py:205]: So these f-Strings are great! Unfortunately, they are not supported until Python```>=3.6```... Let me think a bit if I want to drop Python 3.5 support.

File: pwlf/pwlf.py [hunk 2]
```diff
@@ -238,6 +227,44 @@ def __init__(self, x, y, disp_res=False, lapack_driver='gelsd', degree=1,
         self.intercepts = None
         self.se = None
 
+    @staticmethod
```
Comment (cjekel) [inline pwlf/pwlf.py:230]: Thanks! These are very good changes.

**Comment by cjekel** [comment]
Hi Vladimir,

Thanks for the PR. Reducing the number of lines of code is great! and consolidating code is better! Yes, it's probably time to drop support for Python 2.7 :)

Not sure how I feel about dropping support for python 3.5... Let me think about this. In the meantime I'll get the repo ready to depreciate Python 2.7.

Funny thing I had no idea that other repo exisited! Thanks for pointing it to me. Seems like they do a weighted least squares, with assumptions on breakpoints. I know you were interested in weighted least squares methods at one point. 

Cheers!
CJ

**Comment by cjekel** [comment]
@h-vetinari Head's up for conda-forge. I'm planning on releasing a last version for Python 2.7. Then future version will have something like ```python_requires=">3.4"``` in the setup.py. Not sure if that will require a recipe change?

**Comment by h-vetinari** [comment]
> @cjekel: @h-vetinari Head's up for conda-forge. I'm planning on releasing a last version for Python 2.7. Then future version will have something like python_requires=">3.4" in the setup.py. Not sure if that will require a recipe change?

Thanks, that won't be a problem, conda has not been building python 2.7 & 3.5 packages for a while now. The [feedstock](https://github.com/conda-forge/pwlf-feedstock) is currently building for 3.6-3.8 & pypy 3.6.

**Comment by vkhodygo** [comment]
That's good to hear! 
Sorry, I didn't think about v3.5, that's probably a bad habit of mine, I tend to use the latest stable release only. 
Should I add a deprecation warning for the time being?

File: pwlf/pwlf.py [hunk 3]
```diff
@@ -1464,10 +1441,10 @@ def p_values(self, method='linear', step_size=1e-4):
             # calculate my t-value
             t = parameters / self.se
         else:
-            errmsg = "Error: method='" + method + "' is not supported!"
+            errmsg = f"Error: method=\'{method}\' is not supported!"
```
Comment (h-vetinari) [inline pwlf/pwlf.py:1444]: The backslashes shouldn't be necessary here - `'` within `"..."` is safe.

File: pwlf/pwlf.py [hunk 4]
```diff
@@ -1212,14 +1193,14 @@ def standard_errors(self, method='linear', step_size=1e-4):
...
-                A[:, i] = (f-f0) / step_size
+                A[:, i] = (f - f0) / step_size
             e = f0 - self.y_data
             # reset beta and breaks back to original values
             self.beta = orig_beta
             self.fit_breaks = orig_breaks
 
         else:
-            errmsg = "Error: method='" + method + "' is not supported!"
+            errmsg = f"Error: method=\'{method}\' is not supported!"
```
Comment (h-vetinari) [inline pwlf/pwlf.py:1203]: Same here

**Comment by cjekel** [comment]
@vkhodygo 

Can you address h-vetinari's comments above? Otherwise LGTM :)

> Should I add a deprecation warning for the time being?

Not necessary. I think adding the ```python_requires``` on the last release will be good enough, such that everything will still work for someone on an old Python release with ```pip install pwlf==2.0.4```.

> Yeah, I was interested but that part in my thesis proved to be, well... Let's say my examiners do not share my views on the methods I used.

Well you finished now? Congratulations! :tada:

**Comment by vkhodygo** [comment]
Fixed as you asked. 
> Well you finished now? Congratulations! 

Thanks, I'm almost there - finishing some minor corrections as I've passed the viva.

**Comment by cjekel** [comment]
Thanks Vladimir!

@h-vetinari Thank you again for setting up the conda-forge builds and making this work more accessible to other users!

**Pull request merged**